business/services.go

```diff
@@ -214,9 +230,57 @@ func (in *SvcService) buildServiceList(namespace models.Namespace, svcs []core_v
 			Selector:               item.Spec.Selector,
 			IstioReferences:        svcReferences,
 			KialiWizard:            kialiWizard,
+			ServiceRegistry:        "Kubernetes",
 		}
 	}
 
+	// Add Istio Registry Services that are not present in the Kubernetes list
+	rSvcs = kubernetes.FilterRegistryServicesByServices(rSvcs, svcs)
+	for _, item := range rSvcs {
```

Comment:
Can this loop be moved to separate method for reusing porposes?

Reply:
Good point, but I'm not sure, what do you have in mind ?
This loop should be used inside the business/services.go, and other features like validations, appender or security should use the business.SvcService to fetch services (not replicating this logic).

I guess you are thinking on fetching the services from the validations ? Then we could add a AllNamespaces criteria in the ServiceCriteria (or any other needed filtering required for this job).

It's similar to the IstioConfigList service, the `GetIstioConfigList` returns the Istio config, and using the criteria you can fetch information from one source to another.

The goal here is to simplify the pattern and avoid that there are multiple requests for the same logic.

Perhaps, was that your use case ?

Reply:
Yeah, I mean having it as a separate method within the business/services.go, and be also called in a method "GetServiceList" when param AllNamespaces is true. So when AllNamespaces is true, then similar to "GetIstioConfigList" it will list Services from Registry only and will process them, but without code duplication, especially the "ServiceOverview" objects creation from Registry Services.

Reply:
Ah, I guess you mean to have a split of the "buildServiceList" in two methods, one per kubernetes Services and another for Istio Registry Services and combine/concat.

I can try to re-org those methods in that way.

Reply:
right, and to be able to call Registry Services one directly from GetServiceList for AllNamespaces true case